media/org.openttd.OpenTTD.metainfo.xml

```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--Copyright 2018-2020 Flathub maintainers-->
+<component type="desktop-application">
+  <id>org.openttd.OpenTTD.desktop</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0-only</project_license>
+  <name>OpenTTD</name>
+  <summary>Open source simulation game based upon Transport Tycoon Deluxe</summary>
+  <description>
+    <p>
+      OpenTTD is an open source simulation game based upon the popular
+      Microprose game "Transport Tycoon Deluxe", written by Chris Sawyer. It
+      attempts to mimic the original game as closely as possible while
+      extending it with new features.
+    </p>
+  </description>
+  <launchable type="desktop-id">openttd.desktop</launchable>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://www.openttd.org/screenshots/1.4-02-opengfx-1920x1200.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://www.openttd.org/screenshots/1.9-darkuk-1.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://www.openttd.org/screenshots/1.9-darkuk-2.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://www.openttd.org/screenshots/1.9-darkuk-3.png</image>
+    </screenshot>
+  </screenshots>
+  <url type="homepage">https://www.openttd.org/</url>
+  <url type="bugtracker">https://github.com/OpenTTD/OpenTTD/issues</url>
+  <url type="help">https://wiki.openttd.org/en/Manual/</url>
+  <developer_name>The OpenTTD team</developer_name>
+  <update_contact>info@openttd.org</update_contact>
+  <provides>
+    <binary>openttd</binary>
+  </provides>
+  <releases>
+    <release version="13.4" date="2023-08-05"/>
+    <release version="13.3" date="2023-06-11"/>
+    <release version="13.2.1" date="2023-06-11"/>
+    <release version="13.2" date="2023-06-10"/>
+    <release version="13.1" date="2023-04-10"/>
+    <release version="13.0" date="2023-02-05"/>
+    <release version="12.2" date="2022-04-02"/>
+    <release version="12.1" date="2021-11-08"/>
+    <release version="12.0" date="2021-10-17"/>
+    <release version="1.11.2" date="2021-05-03"/>
+    <release version="1.11.1" date="2021-04-18"/>
+    <release version="1.11.0" date="2021-04-01"/>
+    <release version="1.10.3" date="2020-08-09"/>
+    <release version="1.10.2" date="2020-06-01"/>
+    <release version="1.10.1" date="2020-04-13"/>
+    <release version="1.10.0" date="2020-04-01"/>
+  </releases>
```

Comment:
This `<release>` blob is impractical for us to maintain. Please remove it.

Reply:
It might be possible to auto-fill it by parsing changelog when configuring/building

Reply:
Please don't. It will break. We will not find out :) We have been around this bush a few times already ;)

Reply:
Understood, having extra release tags is impractical. But what if we had a tool to generate it instead? Just like shell script:

```
#!/bin/sh
# Generate metainfo releases from git tags
#

# <release version="1.10.1" date="2020-04-13"/>
#
VERSIONS=${1:-10}

get_date() {
	git log -1 --pretty=format:%as $@
}

get_tags() {
	git tag -l | grep -Ev -- '-(RC|beta).*' | sort -r --version-sort
}

print_release() {
	local VER="$1"
	DATE=$(get_date $VER)
	echo "<release version=\"$VER\" date=\"$DATE\"/>"
}

if [ -n "$VERSIONS" ] && [ "$VERSIONS" != all ]; then
	get_tags | head -${VERSIONS} | while read VER; do print_release $VER; done
else
	get_tags | while read VER; do print_release $VER; done
fi
```

Would that be acceptable instead?

Reply:
The issue here too is that we have to maintain these things. And we are having a hard time as it is to maintain our current release version etc. We often even forget to the the simplest thing upon release, causing issues like 13.2 / 13.3 / 13.4.

Especially for files that won't show up in a short period of time after release that they are borked, it is very annoying to throw any automation against it.

But maybe more of a question: what value does it add to see the version history in a file like this? If it is just to help GNOME etc to show OpenTTD, why would it want to know the history? (honest question, to be clear)

Reply:
Btw, if you really need an entry in the releases list, you could do something similar as with `openttd.desktop` (which needs the actual binary name), and have a single entry which is replaced with the current version the game is building for.

This would be a bit of work, as you need to do something similar for `openttd.desktop` in `media/CMakeLists.txt`, and update `cmake/scripts/Desktop.cmake` to also be able to replace the version. But that way you could have the latest version in the list there, in an automated way, that doesn't require any real maintenance from our side.

But if we can just not have the list, that would work too. Fully depends on what it is actually used for :)

Reply:
Yes, at least one release record is mandatory for strictly validated metainfo file. Problem is we need also the release date, not only current version. Having only single record would be a bit poor choice, but better than nothing at all. Ideally the release tag should contain short summary about what is new in that version.

Reply:
Short summary we don't have. Like, literally, we don't have it :P We make a short news post poking fun of things, but that is about as far as it gets :D

There is `REV_ISODATE`, which is the date of the current build you are running. For a release that will be the date of the release.